test/tests/functional/pbs_multi_sched.py

```diff
@@ -2195,3 +2215,393 @@ def test_multi_sched_priority_sockets(self):
+        self.common_setup2()
+        # Turn off scheduling in all schedulers but one (say sc3)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'false'},
+                            id="sc1")
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'false'},
+                            id="sc2")
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'false'},
+                            id="default")
+        a = {'Resource_List.select': '1:ncpus=2', 'reserve_start': int(
+            time.time() + 5), 'reserve_end': int(time.time() + 15)}
+        r = Reservation(TEST_USER, a)
+        rid = self.server.submit(r)
+        a = {'reserve_state': (MATCH_RE, 'RESV_RUNNING|5')}
+        self.server.expect(RESV, a, rid)
+        self.logger.info("Wait for reservation to end")
+        time.sleep(10)
+        msg = "Resv partition is P3"
+        self.server.log_match(msg)
+
+    def test_setting_partition_resv_hook(self):
+        """
+        Test if a reservation can set partition name on reservation object
+        """
+        hook_body = """
+import pbs
+e = pbs.event()
+resv = e.resv
+resv.partition = "P-3"
+pbs.logmsg(pbs.EVENT_DEBUG, "Resv partition is %s" %resv.partition)
+e.accept()
+"""
+        a = {'event': 'resvsub', 'enabled': 'true', 'debug': 'true'}
+        self.server.create_import_hook("h1", a, hook_body)
+        # Create 3 multi-scheds sc1, sc2 and sc3, 4 partitions and 4 vnodes
+        self.common_setup2()
+        a = {'Resource_List.select': '1:ncpus=2', 'reserve_start': int(
+            time.time() + 5), 'reserve_end': int(time.time() + 15)}
+        r = Reservation(TEST_USER, a)
+        try:
+            rid = self.server.submit(r)
+        except PbsSubmitError as e:
+            msg = "resv attribute 'partition' is readonly"
+            self.server.log_match(msg)
+            self.assertTrue("hook 'h1' encountered an exception" in e.msg[0])
+
+    def test_resv_alter(self):
+        """
+        Test if a reservation confirmed by a multi-sched can be altered by the
+        same scheduler.
+        """
+        self.common_setup2()
+        # Submit 4 reservations to fill up the system and check they are
+        # confirmed
+        for _ in range(4):
+            a = {'Resource_List.select': '1:ncpus=2', 'reserve_start': int(
+                time.time() + 60), 'reserve_end': int(time.time() + 120)}
+            r = Reservation(TEST_USER, a)
+            rid = self.server.submit(r)
+            attr = {'reserve_state': (MATCH_RE, 'RESV_CONFIRMED|2')}
+            self.server.expect(RESV, attr, rid)
+            partition = self.server.status(RESV, 'partition', id=rid)
+            if (partition[0]['partition'] == 'P4'):
+                    old_end_time = a['reserve_end']
+                    modify_resv = rid
+        # Modify the endtime of reservation confirmed on partition P4 and
+        # make sure the node solution is correct.
+        end_time = old_end_time + 60
```

Comment:
since there are 4 reservations submitted to run on all configured partitions, one of the reservation must be confirmed on P4. It can only fail if there is a bug in scheduler and it does not confirm a reservation.